### PR TITLE
GEOPY-719

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-719 - Run batch ip 2d inversion**

### CODE REVIEW

File: geoapps/inversion/electricals/induced_polarization/pseudo_three_dimensions/driver.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    def __init__(
+        self, params: InducedPolarizationPseudo3DParams
+    ):  # pylint: disable=W0235
+        super().__init__(params)
+        if params.files_only:
+            sys.exit("Files written")
+
+    def write_files(self, lookup):
+        """Write ui.geoh5 and ui.json files for sweep trials."""
```
Comment (domfournier) [inline geoapps/inversion/electricals/induced_polarization/pseudo_three_dimensions/driver.py:47]: Docstring input

File: geoapps/inversion/electricals/induced_polarization/pseudo_three_dimensions/params.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    forward_ui_json,
+    inversion_defaults,
+    inversion_ui_json,
+    validations,
+)
+
+
+class InducedPolarizationPseudo3DParams(InversionBaseParams):
+    """
+    Parameter class for electrical->conductivity inversion.
```
Comment (domfournier) [inline geoapps/inversion/electricals/induced_polarization/pseudo_three_dimensions/params.py:25]: Change description for IP

File: geoapps/inversion/line_sweep/driver.py [hunk 3]
```diff
@@ -87,15 +90,55 @@ def collect_results(self):
...
-            mesh.name = f"Line {line}"
+            with Workspace(f"{os.path.join(path, files[line])}.ui.geoh5") as ws:
+                survey = ws.get_entity("Data")[0]
+                data = self.collect_line_data(survey, data)
+                mesh = ws.get_entity("Models")[0]
+                mesh = mesh.copy(parent=models_group)
+                mesh.name = f"Line {line}"
+                drape_models.append(mesh)
 
         data_result.add_data(data)
```
Comment (domfournier) [inline geoapps/inversion/line_sweep/driver.py:103]: Hmmm, this would have to be inside the context to work...

File: geoapps/utils/models.py [hunk 4]
```diff
@@ -12,14 +12,108 @@
...
+#                     f"Found more than one data set with name {names[ind]} in"
+#                     f"model {model.name}."
+#                 )
+#             octree_model[lookup_inds] = datum[0].values
+#
+#         octree_model = octree_model[mesh._ubc_order]
+#         octree_model[~active] = np.nan
+#         octree.add_data({label: {"values": octree_model}})
+#
+#     return octree
```
Comment (domfournier) [inline geoapps/utils/models.py:51]: Remove commented code

File: geoapps/utils/models.py [hunk 5]
```diff
@@ -12,14 +12,108 @@
...
+
+def drape_to_octree(
+    octree: Octree,
+    drape_model: DrapeModel | list[DrapeModel],
+    children: dict[str, list[str]],
+    active: np.ndarray,
+    method: str = "lookup",
+):
+    """
+    Interpolate drape model(s) into octree mesh.
```
Comment (domfournier) [inline geoapps/utils/models.py:62]: Need docstrings for inputs

**Review by domfournier — COMMENTED** [review]
Should be straight forward. Copilot might help ;)

File: geoapps/inversion/electricals/induced_polarization/pseudo_three_dimensions/driver.py (hunk 1 above)
Comment (benk-mira) [reply geoapps/inversion/electricals/induced_polarization/pseudo_three_dimensions/driver.py:47]: Done

File: geoapps/inversion/electricals/induced_polarization/pseudo_three_dimensions/params.py (hunk 2 above)
Comment (benk-mira) [reply geoapps/inversion/electricals/induced_polarization/pseudo_three_dimensions/params.py:25]: Done

File: geoapps/inversion/line_sweep/driver.py (hunk 3 above)
Comment (benk-mira) [reply geoapps/inversion/line_sweep/driver.py:103]: Don't think so.  The above context is for reading the line workspaces, so I can pile up results in the data which is then added to data_result whose workspace is self.workspace (still open in the context started at line33)

File: geoapps/utils/models.py (hunk 4 above)
Comment (benk-mira) [reply geoapps/utils/models.py:51]: Done

File: geoapps/utils/models.py (hunk 5 above)
Comment (benk-mira) [reply geoapps/utils/models.py:62]: Done

File: geoapps/inversion/line_sweep/driver.py (hunk 3 above)
Comment (domfournier) [reply geoapps/inversion/line_sweep/driver.py:103]: Got it, different workspace.

**Review by domfournier — DISMISSED** [review]
image

**Review by domfournier — APPROVED** [review]

**Pull request merged**